### PR TITLE
fix(pipeline): skip cancel if worker not found logic task due to user frequent click

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
skip cancel if worker not found logic task due to user frequent click

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?id=337650&iterationID=1389&pId=0&type=BUG)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |  Fix the bug that skip cancel if worker not found logic task due to user frequent click            |
| 🇨🇳 中文    |    修复了由于用户频繁点击取消造成流水线panic的问题          |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: internal/tools/pipeline/providers/leaderworker/impl_worker.go [hunk 1]
```diff
@@ -244,10 +244,14 @@ func (p *provider) workerListenCancelingLogicTask(ctx context.Context, w worker.
 			p.Log.Infof("logic task cancel request received, begin cancel it, workerID: %s, logicTaskID: %s", w.GetID(), logicTaskID)
 			// do cancel
 			p.lock.Lock()
-			tctx := p.forWorkerUse.myWorkers[w.GetID()].LogicTasks[logicTaskID].Ctx
-			lwctx.IdempotentCloseTaskCancelChan(tctx)
+			defer p.lock.Lock()
```
Comment (sfwn) [inline internal/tools/pipeline/providers/leaderworker/impl_worker.go:247]: should be unlock, my friend.

Comment (sfwn) [reply internal/tools/pipeline/providers/leaderworker/impl_worker.go:247]: test it please.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/tools/pipeline/providers/leaderworker/impl_worker.go (hunk 1 above)
Comment (chengjoey) [reply internal/tools/pipeline/providers/leaderworker/impl_worker.go]: done

**Review by sfwn — APPROVED** [review]

**Pull request merged**